### PR TITLE
fix(opt): order derive of batch scan

### PR DESCRIPTION
**Pull request by st1page**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
Now during derivation of batch scan is wrong and not just use prefix. if a table's order column is (a,b,c,d) and output indices is (a,c,e) then it will derive the (a,c). that's wrong.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

**Review by xxchan — COMMENTED** [review]
Could you explain a bit what the problem is?

File: src/frontend/src/optimizer/plan_node/logical_scan.rs [hunk 1]
```diff
@@ -223,28 +223,34 @@ impl LogicalScan {
...
+            self.table_desc().columns.len(),
+        )
+    }
+
     /// Return indices of fields the output is ordered by and
     /// corresponding direction
     pub fn get_out_column_index_order(&self) -> Order {
-        let id_to_op_idx = Self::get_id_to_op_idx_mapping(&self.output_col_idx, &self.table_desc);
-        Order::new(
+        let id_to_tb_idx = self.table_desc.get_id_to_op_idx_mapping();
```
Comment (xxchan) [inline src/frontend/src/optimizer/plan_node/logical_scan.rs:237]: Is `Self::get_id_to_op_idx_mapping` still useful, and what's their difference? Maybe add some comments

Comment (st1page) [reply src/frontend/src/optimizer/plan_node/logical_scan.rs:237]: I think they are both useful.

Comment (st1page) [reply src/frontend/src/optimizer/plan_node/logical_scan.rs:237]: here we need an internal order before `output_indices` to use the `ColumnIndexMapping::rewrite_provide_order`

**Review by chenzl25 — APPROVED** [review]
LGTM

File: src/frontend/src/optimizer/plan_node/logical_scan.rs (hunk 1 above)
Comment (xxchan) [reply src/frontend/src/optimizer/plan_node/logical_scan.rs:237]: I think we can have `self.table_desc.order()`

Comment (st1page) [reply src/frontend/src/optimizer/plan_node/logical_scan.rs:237]: a little strange to me and I think this is the BatchScan's behavior

Comment (st1page) [reply src/frontend/src/optimizer/plan_node/logical_scan.rs:237]: and for the `get_id_to_op_idx_mapping` should also be used on distribution and when we fix https://github.com/singularity-data/risingwave/blob/main/src/frontend/src/optimizer/plan_node/batch_seq_scan.rs#L96 later.

**Pull request merged**